src/objects/zcl_abapgit_object_avas.clas.abap

```diff
@@ -0,0 +1,239 @@
+class ZCL_ABAPGIT_OBJECT_AVAS definition
+  public
+  inheriting from ZCL_ABAPGIT_OBJECTS_SUPER
+  create public .
+
+public section.
+
+  interfaces ZIF_ABAPGIT_OBJECT .
+PROTECTED SECTION.
+
+  TYPES: BEGIN OF ty_header,
+           guid      TYPE guid_32,
+           attribute TYPE cls_attribute_name,
+           object    TYPE pak_object_key,
+         END OF ty_header.
+
+  TYPES: BEGIN OF ty_avas,
+           header TYPE ty_header,
+           values TYPE cls_value_assignments,
+           links  TYPE cls_linked_objects,
+         END OF ty_avas.
+
+  METHODS instantiate
+    RETURNING
+      VALUE(ro_avas) TYPE REF TO cl_cls_attr_value_assignment .
+private section.
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_OBJECT_AVAS IMPLEMENTATION.
+
+
+  METHOD instantiate.
+
+    DATA: lv_id TYPE guid_32.
+
+
+    lv_id = ms_item-obj_name.
+
+    TRY.
+        CREATE OBJECT ro_avas
+          EXPORTING
+*           im_object        =
+*           im_attribute     =
+            im_assignment_id = lv_id
+*           im_package       =
+*           im_cache         =
+*           im_report_run_result    =
+          .
+      CATCH cx_pak_wb_object_locked.
+        BREAK-POINT.
+      CATCH cx_pak_not_authorized.
+        BREAK-POINT.
+      CATCH cx_pak_invalid_state.
+        BREAK-POINT.
+      CATCH cx_pak_invalid_data.
+        BREAK-POINT.
+    ENDTRY.
+
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+
+    DATA: lo_avas TYPE REF TO cl_cls_attr_value_assignment.
+
+
+    lo_avas = instantiate( ).
+
+    lo_avas->if_pak_wb_object~get_last_changed(
+      IMPORTING
+        ex_changed_by = rv_user ).
+
+    IF rv_user IS INITIAL.
+      rv_user = c_user_unknown.
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~compare_to_remote_version.
+    CREATE OBJECT ro_comparison_result TYPE zcl_abapgit_comparison_null.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lo_avas TYPE REF TO cl_cls_attr_value_assignment.
+
+
+    lo_avas = instantiate( ).
+
+    TRY.
+        lo_avas->if_cls_attr_value_assignment~lock_and_refresh(
+          im_allow_popups = abap_false ).
+      CATCH cx_pak_invalid_state
+          cx_pak_invalid_data
+          cx_pak_not_authorized
+          cx_pak_wb_object_locked.
+        BREAK-POINT.
+    ENDTRY.
+
+    lo_avas->if_pak_wb_object~delete( ).
+
+    lo_avas->if_pak_wb_object~save( ).
+
+    lo_avas->if_pak_wb_object_internal~unlock( ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+
+    DATA: lo_avas TYPE REF TO cl_cls_attr_value_assignment,
+          ls_avas TYPE ty_avas.
+
+
+    io_xml->read( EXPORTING iv_name = 'AVAS'
+                  CHANGING cg_data = ls_avas ).
+
+    tadir_insert( iv_package ).
+
+* todo, this creates a new GUID
```

Comment:
Does that mean that after pull we will see a difference?

Reply:
no, above code is not in the final code, https://github.com/larshp/abapGit/blob/master/src/objects/zcl_abapgit_object_avas.clas.abap